tests/Unit/CacheCheckerTest.php

```diff
@@ -0,0 +1,271 @@
+<?php
+
+/**
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; under version 2
+ * of the License (non-upgradable).
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * Copyright (c) 2020 (original work) Open Assessment Technologies SA;
+ */
+
+declare(strict_types=1);
+
+namespace OAT\Library\HealthCheckCache\Tests\Unit;
+
+use Exception;
+use OAT\Library\HealthCheckCache\CacheChecker;
+use OAT\Library\HealthCheckCache\CacheKeyGeneratorInterface;
+use PHPUnit\Framework\MockObject\MockObject;
+use PHPUnit\Framework\TestCase;
+use Psr\Cache\CacheException;
+use Psr\Cache\CacheItemInterface;
+use Psr\Cache\CacheItemPoolInterface;
+
+final class CacheCheckerTest extends TestCase
+{
+    private const CUSTOM_KEY = 'test-key';
+
+    /** @var CacheChecker */
+    private $subject;
+
+    /** @var CacheItemPoolInterface|MockObject */
+    private $cachePoolMock;
+
+    /** @var CacheKeyGeneratorInterface|MockObject */
+    private $keyGeneratorMock;
+
+    protected function setUp(): void
+    {
+        $this->cachePoolMock = $this->createMock(CacheItemPoolInterface::class);
+        $this->keyGeneratorMock = $this->createMock(CacheKeyGeneratorInterface::class);
+
+        $this->subject = new CacheChecker($this->cachePoolMock, $this->keyGeneratorMock);
+    }
+
+    public function testGetIdentifier(): void
+    {
+        self::assertEquals('cache', $this->subject->getIdentifier());
+    }
+
+    public function testForSuccessCheckerResult(): void
+    {
+        $cacheItemMock = $this->createMock(CacheItemInterface::class);
```

Comment:
using array cache as suggested would have actually avoided this long list of mock settings :)

Reply:
Yes, I wanted to used that but since there is a delete at the end of the process, I can only test whether the item is no more in the cache, right? Or am I thinking in a wrong way?

Reply:
- real cache implementation usage (like the array one) will fail on I/O errors, so your test will see it
- but if you feel more confortable testing each expected cache call via mocks, thats also totally valid for me

I just brought this for test readability, but your proposal is nevertheless robust